### ISSUE TITLE
jsx-no-literals with allowStrings doesn't work in props

### ISSUE THREAD

**Issue by e110c0** [yannickcr/eslint-plugin-react#2720]
the following setup seems not to work:

```js
<button type="button" ></button>
```

with the rule 
```js
    'react/jsx-no-literals': [
      'warn',
      {
        noStrings: true,
        allowedStrings: ['button', 'submit'],
      },
    ],
```

produces the following error: `Invalid prop value: “type="button"”  react/jsx-no-literals`

Context: I'm trying to find a balance between enforcing no literals but also `react/button-has-type`

**Comment by ljharb** [comment]
That should definitely work - jsx-no-literals shouldn't be warning on a prop value.

The `ignoreProps` option might help, however.

**Comment by e110c0** [comment]
`ignoreProps` works, in general, but also doesn't check any props at all. It seems that the `allowedStrings` are not checked in  props at all.

**Comment by karolina-benitez** [comment]
I believe I have a solution for this. I will open a PR later today
